projects/core/src/lib/types/mask-processors.ts

```diff
@@ -0,0 +1,8 @@
+import {ElementState} from './element-state';
+
+export type MaskPreprocessor = (_: {elementState: ElementState; data?: string}) => {
+    elementState: ElementState;
+    data?: string;
```

Comment:
I think it would be easier for people not to test it for `undefined` all the time, we can just pass empty string for those cases.